SonarQube.Common/ProcessRunner.cs

```diff
@@ -64,6 +87,16 @@ public bool Execute(string exeName, string args, string workingDirectory, int ti
                 WorkingDirectory = workingDirectory
             };
 
+            if (envVariables != null)
+            {
+                foreach (KeyValuePair<string, string> envVariable in envVariables)
+                {
+                    if (!String.IsNullOrEmpty(envVariable.Value))
```

Comment:
Why exclude null/empty values? The caller might deliberately be choosing to clear a variable.
Null/empty keys are probably more of an issue.